l3kernel/l3file.dtx

```diff
@@ -666,6 +709,40 @@
 %   function. An error is raised if the file is not found.
 % \end{function}
 %
+% \begin{function}[added = 2014-07-02]{\file_if_exist_input:n, \file_if_exist_input:nF}
+%   \begin{syntax}
+%     \cs{file_if_exist_input:n} \Arg{file name}
+%     \cs{file_if_exist_input:nF} \Arg{file name} \Arg{false code}
+%   \end{syntax}
+%   Searches for \meta{file name} using the current \TeX{} search
+%   path and the additional paths controlled by
+%   \cs{file_path_include:n}. If found then
+%   reads in the file as additional \LaTeX{} source as described for
+%   \cs{file_input:n}, otherwise inserts the \meta{false code}.
+%   Note that these functions do not raise
+%   an error if the file is not found, in contrast to \cs{file_input:n}.
+% \end{function}
+%
+% \begin{function}[added = 2017-07-07]{\file_input_stop:}
+%   \begin{syntax}
+%     \cs{file_input_stop:}
+%   \end{syntax}
+%   Ends the reading of a file started by \cs{file_input:n} or similar before
+%   the end of the file is reached. Where the file reading is being terminated
+%   due to an error, \cs{msg_critical:nn(nn)} should be preferred.
```

Comment:
\cs[index=msg_critical:nn]{msg_critical:nn(nn)} or similar to avoid messing up the index.

Reply:
It might be useful to comment in this function why we made it protected rather than expandable (for consistency with other `file` functions, or a similar reason).